src/oemof/solph/components/_link.py

```diff
@@ -71,14 +78,30 @@ class Link(on.Transformer):
     0.8
     """
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
+    def __init__(
+        self,
+        label=None,
+        inputs=None,
+        outputs=None,
+        conversion_factors=None,
+    ):
+        if inputs is None:
+            warn_if_missing_attribute(self, "inputs")
+            inputs = {}
+        if outputs is None:
+            warn_if_missing_attribute(self, "outputs")
+            outputs = {}
+        if conversion_factors is None:
+            warn_if_missing_attribute(self, "conversion_factors")
+            conversion_factors = {}
```

Comment:
Wouldn't it be easier to just require the arguments instead of having None as default? Whoever wants None can still give it manually.

Reply:
I totally agree, especially as they are necessary in inputs. Thanks for the hint :)